core/forms.js

```diff
@@ -2,7 +2,9 @@ define('core/forms', ['core/z'], function(z) {
 
     function checkValid(form) {
         if (form) {
-            $(form).filter(':not([novalidate])').find('button[type=submit]').attr('disabled', !form.checkValidity());
+            var $currentForm = $(form);
```

Comment:
As I mentioned on IRC we should do `var $form = $(form), $textarea = $form.find('textarea');` and use those later.